autocodesign/localcodesignasset/profilelookup.go

```diff
@@ -0,0 +1,121 @@
+package localcodesignasset
+
+import (
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/bitrise-io/go-xcode/autocodesign"
+	"github.com/bitrise-io/go-xcode/profileutil"
+)
+
+func findProfile(localProfiles []profileutil.ProvisioningProfileInfoModel, platform autocodesign.Platform, distributionType autocodesign.DistributionType, bundleID string, entitlements autocodesign.Entitlements, minProfileDaysValid int, certSerials []string, deviceIDs []string) *profileutil.ProvisioningProfileInfoModel {
+	for _, profile := range localProfiles {
+		if isProfileMatching(profile, platform, distributionType, bundleID, entitlements, minProfileDaysValid, certSerials, deviceIDs) {
+			return &profile
+		}
+	}
+
+	return nil
+}
+
+func isProfileMatching(profile profileutil.ProvisioningProfileInfoModel, platform autocodesign.Platform, distributionType autocodesign.DistributionType, bundleID string, entitlements autocodesign.Entitlements, minProfileDaysValid int, certSerials []string, deviceIDs []string) bool {
+	if !isActive(profile, minProfileDaysValid) {
+		return false
+	}
+
+	if !hasMatchingDistributionType(profile, distributionType) {
+		return false
+	}
+
+	if !hasMatchingBundleID(profile, bundleID) {
+		return false
+	}
+
+	if !hasMatchingPlatform(profile, platform) {
+		return false
+	}
+
+	if !hasMatchingLocalCertificate(profile, certSerials) {
+		return false
+	}
+
+	if !containsAllAppEntitlements(profile, entitlements) {
+		return false
+	}
+
+	if !provisionsDevices(profile, deviceIDs) {
+		return false
+	}
+
+	return true
+}
+
+func hasMatchingBundleID(profile profileutil.ProvisioningProfileInfoModel, bundleID string) bool {
+	return profile.BundleID == bundleID
+}
+
+func hasMatchingLocalCertificate(profile profileutil.ProvisioningProfileInfoModel, localCertificateSerials []string) bool {
```

Comment:
What if there is a matching local certifiacte, but it is not used in the autocodesign.AppCodesignAssets struct in the end? Do we check that all profiles contains the same cert in the end?

Reply:
In case of certificates the `localcodesignassets` package is not working with the local certificates, but the once provided by `selectCertificatesAndDistributionTypes`: https://github.com/bitrise-io/go-xcode/pull/135/files#diff-2e5388add0be9211c62768963248e293009baa4e868382cb0e5575c8a56e96faR156

and: https://github.com/bitrise-io/go-xcode/pull/135/files#diff-2e5388add0be9211c62768963248e293009baa4e868382cb0e5575c8a56e96faR183

Reply:
Decided to require all local certificates to be included, as that's how EnsureCodesignAssets also works.